cram/cram_encode.c

```diff
@@ -3150,7 +3164,9 @@ int cram_put_bam_seq(cram_fd *fd, bam_seq_t *b) {
          * multiple sequences per container we emit the small partial
          * container as-is and then start a fresh one in a different mode.
          */
-        if (multi_seq) {
+        if (multi_seq == 0 && fd->multi_seq == 1 && fd->multi_seq_user == -1) {
+            fd->multi_seq = -1;
+        } else if (multi_seq || fd->multi_seq == 1) {
```

Comment:
These two conditions can both be true at the same time. Is this the intended behaviour?